contracts/src/v0.8/dev/vrf/SubscriptionAPI.sol

```diff
@@ -230,17 +230,30 @@ abstract contract SubscriptionAPI is ConfirmedOwner, ReentrancyGuard, ERC677Rece
     public
     view
     override
-    returns (uint96 balance, uint96 ethBalance, uint64 reqCount, address owner, address[] memory consumers)
+    returns (
+      uint96 balance,
+      uint96 ethBalance,
+      uint64 reqCount,
+      address owner,
+      address[] memory consumers,
+      uint64[] memory nonces
+    )
   {
     if (s_subscriptionConfigs[subId].owner == address(0)) {
       revert InvalidSubscription();
     }
+    consumers = s_subscriptionConfigs[subId].consumers;
+    nonces = new uint64[](consumers.length);
+    for (uint256 i = 0; i < consumers.length; i++) {
+      nonces[i] = s_consumers[consumers[i]][subId];
```

Comment:
ah one potential I see with this approach is that if consumer is removed, we will lose nonces for those consumers. So the sum of nonces minus total requests can get negative I believe.

Reply:
That's a good note, that can (and probably will) definitely happen since people remove consumers pretty frequently.

Reply:
In that case, if this is something only for UI and doesn't have to be exact, maybe the approximate approach for calculating # of pending requests (that is not super re-org proof) is best? On every `RandomWordsRequested` increment a number in a table and on every `RandomWordsFulfilled` for the same sub decrement?

Reply:
## Issue

- If consumer is removed, we will lose nonces for those consumers
- So the sum of consumer nonces does not reflect total number of reqs served by a subscription anymore.

## Solutions

**Approach 1**

- Add another entry in s_consumers mapping for each subId, as a global nonce counter for a given subscription ID.

```
 s_consumers[0][subId] - Represents global nonce for a given subId across all consumers 
```

- This means we've to increment two counter for each request, which means more gas cost in hot path

```
s_consumers[consumerId][subId]++
s_consumers[0][subId]++
```

**Approach 2**

- Do not delete consumer. Just mark it as inactive in the s_consumers mapping.
- This may have some issues with MAX_CONSUMERS semantics. Customers may not be able to add newer consumers after reaching MAX_CONSUMERS and even after deleting some existing consumers then.

These approaches may not work, unless we add gas cost to hot path.